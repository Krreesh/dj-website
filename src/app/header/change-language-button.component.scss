@import "../../styles";

.dropdown-item {
  padding: .25rem 1.25rem;
}
