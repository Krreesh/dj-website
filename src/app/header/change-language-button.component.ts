import {Component, Input} from '@angular/core';
import {LanguageService} from '../translation/language.service';

@Component({
  selector: 'app-change-language-button',
  styleUrls: ['./change-language-button.component.scss'],
  template: `
    <button class="dropdown-item changeLanguage" (click)="useLanguage(language)">{{language}}</button>
  `
})
export class ChangeLanguageButtonComponent {

  @Input()
  language: string;

  constructor(private languageService: LanguageService) {
  }

  useLanguage(language: string): void {
    this.languageService.updateCurrentLanguage(language);
  }
}
