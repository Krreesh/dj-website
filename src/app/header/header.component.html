<header class="header">
  <nav class="navbar navbar-expand-lg navbar-light">
        <span class="navbar-brand text">
          Dieter Jordens
        </span>
    <button class="navbar-toggler toggleNavBar"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            [attr.aria-expanded]="!isCollapsed"
            aria-label="Toggle navigation"
            (click)="toggle()">
      <span class="navbar-toggler-icon"></span>
    </button>
    <div class="collapse navbar-collapse" id="navbarSupportedContent" [ngbCollapse]="isCollapsed">
      <app-nav-options class="mr-auto"
                       (navigated)="closeNavigationBar()"></app-nav-options>
      <app-language-selector></app-language-selector>
    </div>
  </nav>
</header>
