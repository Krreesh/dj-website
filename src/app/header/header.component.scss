@import "../../colors";

.hamburger, .hamburger.is-active {
  .hamburger-inner{
    background-color: white;
  }
  .hamburger-inner::before{
    background-color: white;
  }
  .hamburger-inner::after{
    background-color: white;
  }
}

.language-selector{
  position: absolute;
  top: 0.5em;
  right: 0.5em;
}

.theme {
  color: #FFFFFF;
  background-color: #398AD7;
}

.navbar-brand:hover, .navbar-brand:visited, .navbar-brand:active, .navbar-brand:focus {
  color: #FFFFFF;
}

.hamburger {
  outline: unset;
}

.displayRight{
  right: 0;
}

@media (min-width: 1001px) {
  .hidden-desktop {
    display:none;
  }
}

@media (max-width: 1000px) {
  .hidden-mobile {
    display: none;
  }
}
