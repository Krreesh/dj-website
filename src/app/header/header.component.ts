import {Component} from '@angular/core';

@Component({
  selector: 'app-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.scss']
})
export class HeaderComponent {
  isCollapsed = true;

  toggle(): void {
    this.isCollapsed = !this.isCollapsed;
  }

  closeNavigationBar(): void {
    this.isCollapsed = true;
  }
}
