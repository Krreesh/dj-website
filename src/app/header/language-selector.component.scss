@import "../../styles.scss";

.dropdown-menu {
  min-width: 0;
}

#languageSelectorDropdown {
  color: $black-color;
}
