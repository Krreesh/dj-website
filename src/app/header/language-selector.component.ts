import {Component} from '@angular/core';
import {LanguageService} from '../translation/language.service';
import {Observable} from 'rxjs';
import {Languages} from '../translation/languages';

@Component({
  selector: 'app-language-selector',
  template: `
    <ng-container *ngIf="currentAndOtherLanguage$ | async as languages">
      <div ngbDropdown class="d-inline-block dropdown">
        <button class="btn btn-outline-dark dropdown-toggle changeLanguageButton" id="languageSelectorDropdown"
                ngbDropdownToggle>{{languages.current | uppercase}}</button>
        <div ngbDropdownMenu aria-labelledby="languageSelectorDropdown" class="dropdown-menu">
          <app-change-language-button *ngFor="let otherLanguage of languages.other" [language]="otherLanguage"></app-change-language-button>
        </div>
      </div>
    </ng-container>
  `,
  styleUrls: ['./language-selector.component.scss']
})
export class LanguageSelectorComponent {
  currentAndOtherLanguage$: Observable<Languages>;

  constructor(private languageService: LanguageService) {
    this.currentAndOtherLanguage$ = this.languageService.getCurrentAndOtherLanguages$();
  }
}
