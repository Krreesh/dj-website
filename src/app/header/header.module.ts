import {NgModule} from '@angular/core';
import {LanguageSelectorComponent} from './language-selector.component';
import {ChangeLanguageButtonComponent} from './change-language-button.component';
import {HeaderComponent} from './header.component';
import {NavOptionsComponent} from '../pages/vision/nav.options.component';
import {CommonModule} from '@angular/common';
import {TranslationModule} from '../translation/translation.module';
import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
import {AppRoutingModule} from '../app-routing.module';

@NgModule({
  declarations: [
    LanguageSelectorComponent,
    ChangeLanguageButtonComponent,
    HeaderComponent,
    NavOptionsComponent
  ],
  exports: [
    HeaderComponent
  ],
  imports: [
    CommonModule,
    TranslationModule,
    NgbModule,
    AppRoutingModule
  ]
})
export class HeaderModule {

}
