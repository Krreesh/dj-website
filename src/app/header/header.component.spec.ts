import {ComponentFixture, TestBed} from '@angular/core/testing';
import {HeaderComponent} from './header.component';
import {MockComponent} from 'ng-mocks';
import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
import {NavOptionsComponent} from './nav-options/nav.options.component';
import {NoopAnimationsModule} from '@angular/platform-browser/animations';

describe('HeaderComponent', () => {
  let component: HeaderComponent;
  let fixture: ComponentFixture<HeaderComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [NgbModule, NoopAnimationsModule],
      declarations: [HeaderComponent, MockComponent(NavOptionsComponent)]
    });
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(HeaderComponent);
    component = fixture.debugElement.componentInstance;
  });

  it('navigation bar is collapsed by default', () => {
    fixture.detectChanges();
    expect(component.collapse).toBe(true);
  });
});
