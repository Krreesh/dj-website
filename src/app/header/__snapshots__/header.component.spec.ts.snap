// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent navigation bar is collapsed by default 1`] = `
<app-header
  isCollapsed={[Function Boolean]}
>
  <header
    class="header"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light"
    >
      <span
        class="navbar-brand text"
      >
         Dieter Jordens 
      </span>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler toggleNavBar"
        data-target="#navbarSupportedContent"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <app-nav-options
          class="mr-auto"
        />
        <app-language-selector />
      </div>
    </nav>
  </header>
</app-header>
`;

exports[`HeaderComponent the toggle button should allow to undo collapse 1`] = `
<app-header
  isCollapsed="false"
>
  <header
    class="header"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light"
    >
      <span
        class="navbar-brand text"
      >
         Dieter Jordens 
      </span>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="true"
        aria-label="Toggle navigation"
        class="navbar-toggler toggleNavBar"
        data-target="#navbarSupportedContent"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse show"
        id="navbarSupportedContent"
      >
        <app-nav-options
          class="mr-auto"
        />
        <app-language-selector />
      </div>
    </nav>
  </header>
</app-header>
`;
